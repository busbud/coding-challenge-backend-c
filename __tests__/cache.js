const { get, set, clear, fromCacheOr } = require("../utils/cache");

describe("Cache", () => {
  beforeAll(() => {
    clear();
  });
  it("should set a value using a custom seriaizer", () => {
    set("key", { a: 1 });
    expect(get("key")).toEqual({ json: { a: 1 }, serialized: '{"a":1}' });
    set("key2", { a: 2 }, value => value.a * 2);
    expect(get("key2")).toEqual({ json: 4, serialized: 4 });
    clear();
    expect(get("key")).toMatchSnapshot();
  });

  it("should cache a given operation or execute it the first time", () => {
    const op = jest.fn(() => ({ a: 1 }));
    fromCacheOr("key", op);
    expect(op).toBeCalledTimes(1);
    const result = fromCacheOr("key", op);
    expect(op).toBeCalledTimes(1);
    expect(result).toMatchSnapshot();
  });
});
