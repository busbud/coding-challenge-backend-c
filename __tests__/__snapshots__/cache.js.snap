// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cache should cache a given operation or execute it the first time 1`] = `
Object {
  "json": Object {
    "a": 1,
  },
  "serialized": "{\\"a\\":1}",
}
`;

exports[`Cache should set a value using a custom seriaizer 1`] = `
Object {
  "json": undefined,
  "serialized": undefined,
}
`;
